Source/Core/Core/AchievementManager.cpp

```diff
@@ -41,15 +43,50 @@ bool AchievementManager::IsLoggedIn() const
   return m_login_data.response.succeeded;
 }
 
+void AchievementManager::LoadGameByFilenameAsync(const std::string& iso_path,
+                                                 const ResponseCallback& callback)
+{
+  rc_hash_generate_from_file(game_hash, RC_CONSOLE_GAMECUBE, iso_path.c_str());
+  ResponseCallback fetch_game_data_callback = [this, callback](ResponseType r_type) {
+    m_is_game_loaded = (r_type == ResponseType::SUCCESS);
+    callback(r_type);
+  };
+  ResponseCallback start_session_callback = [this, fetch_game_data_callback](ResponseType r_type) {
+    if (r_type == ResponseType::SUCCESS)
+      m_queue.EmplaceItem(
+          [this, fetch_game_data_callback] { fetch_game_data_callback(FetchGameData()); });
+  };
+  ResponseCallback resolve_hash_callback = [this, start_session_callback](ResponseType r_type) {
+    if (r_type == ResponseType::SUCCESS)
+      m_queue.EmplaceItem(
+          [this, start_session_callback] { start_session_callback(StartRASession()); });
+  };
+  m_queue.EmplaceItem([this, resolve_hash_callback] { resolve_hash_callback(ResolveHash()); });
+}
+
+void AchievementManager::CloseGame()
+{
+  m_is_game_loaded = false;
+  m_queue.Cancel();
```

Comment:
Why are you clearing outstanding requests here?

Reply:
Safety. If the game is closed, then there's no reason the player should be unlocking achievements for it. I can double check with RetroAchievements coders if we feel it's safe to keep these on the queue until they're processed even if the player has moved on to a new session with a new game, but my understanding so far is that we prefer not to do that.